.github/workflows/ci.yml

```diff
@@ -96,24 +106,46 @@ jobs:
       run: |
         # Do not run updates until GitHub has fixed all the issues: https://github.com/orgs/Homebrew/discussions/4612
         # brew update
-        brew install gcovr ninja || true
+        brew install gcovr || true
 
-    - name: Install Windows Tools
+    - name: Cache OpenCppCoverage
+      if: matrix.type.name == 'Debug' && runner.os == 'Windows'
+      id: opencppcoverage-cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          C:\Program Files\OpenCppCoverage
+        key: opencppcoverage
+
+    - name: Install OpenCppCoverage
       uses: nick-fields/retry@v2
-      if: runner.os == 'Windows'
+      if: matrix.type.name == 'Debug' && runner.os == 'Windows' && steps.opencppcoverage-cache.outputs.cache-hit != 'true'
       with:
         max_attempts: 10
         timeout_minutes: 3
         command: |
-          choco install ninja OpenCppCoverage -y
-          echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
+          choco install OpenCppCoverage -y
+
+    - name: Add OpenCppCoverage to PATH
+      if: matrix.type.name == 'Debug' && runner.os == 'Windows'
+      run: |
+        echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
 
     - name: Install MinGW
       if: matrix.platform.name == 'Windows MinGW'
       run: |
         choco uninstall mingw
         choco install mingw --version 11.2.0
 
+    - name: Cache Mesa 3D
+      if: runner.os == 'Windows'
+      uses: actions/cache@v3
+      with:
+        path: |
+          build/mesa3d-23.0.0-release-msvc/x86
+          build/mesa3d-23.0.0-release-msvc/x64
+        key: mesa3d-23.0.0-release-msvc
```

Comment:
How much time does this save?

Reply:
Hard to say... if you try to compare side by side workflow runs then the timing is all over the place. Having to download and unpack the archive will always take some non-zero amount of time, but if the total time running CMake (discounting the Mesa3D stuff) depends on how generous GitHub is willing to provide us CPU time it is hard to provide any quantifiable value. Percentage-wise, the less time CMake spends doing other stuff the greater this will make a difference, e.g. it provides a bigger benefit when generating for Ninja than for MSBuild.

In the end, the upper limit of time saving is not going to exceed several seconds, so if you think it isn't worth it I can also get rid of the changes.